fairlearn/adversarial/_adversarial_mitigation.py

```diff
@@ -603,7 +603,7 @@ def _validate_input(self, X, y, A, reinitialize=False):
                 dtype=float,
                 allow_nd=True,
             )
-            y = self._validate_data(y, ensure_2d=False)
+            y = self._validate_data(y, dtype=None, ensure_2d=False)
```

Comment:
It seems that this here is an issue. In the case of values like `<=50K` which are string objects, there is an issue when converting them to float [when this line is executed](https://github.com/fairlearn/fairlearn/blob/main/fairlearn/adversarial/_adversarial_mitigation.py#L634). I will look into using OneHotEncoder or `LabelEncoder` to transform these types of values somewhere earlier.